pkg/supervisor/supervisor_test.go

```diff
@@ -210,3 +225,72 @@ func TestSupervisorSnapshotReflectorCtl(t *testing.T) {
 	time.Sleep(100 * time.Millisecond)
 	require.Equal(t, 0, len(reflector.Events))
 }
+
+func TestSupervisorMaximumLedgerLatency(t *testing.T) {
+	tmpPath, err := ioutil.TempDir("", "")
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpPath)
+	ldbDbPath := filepath.Join(tmpPath, "ldb.db")
+	archivePath := filepath.Join(tmpPath, "archive.db")
+
+	svI, err := SupervisorFromConfig(SupervisorConfig{
+		SnapshotInterval: 100 * time.Millisecond,
+		SnapshotURL:      "file://" + archivePath,
+		LDBPath:          ldbDbPath,
+		Reflector:        fakes.NewFakeReflector(),
+		GetLedgerLatency: mockGetLedgerLatency(time.Second, nil),
+		MaxLedgerLatency: time.Minute,
+	})
+	require.NoError(t, err)
+	sv := svI.(*supervisor)
+	defer sv.Close()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	sctx, scancel := context.WithTimeout(ctx, 1*time.Second)
+	defer scancel()
+
+	go func() {
+		// Wait for snapshot to complete
+		time.Sleep(10 * time.Millisecond)
+		// Cancels the context passed to the supervisor, which should cause it
+		// to return from the Start() call
+		scancel()
+	}()
+
+	// This should create a snapshot because the latency is below the max. An archive file should be created.
+	sv.Start(sctx)
+
+	_, err = os.Stat(archivePath)
+	if err != nil {
+		t.Fatalf("No archive created")
+	}
+
+	// Now clear the archive file.
+	err = os.Remove(archivePath)
+	if err != nil {
```

Comment:
In the tests, let's try to use the stretchr test lib.  It makes error checking in tests much more concise, and it's what we use elsewhere in the codebase. Specifically:

```
require.NoErr(t, er)
```